lib/graphql_metrics/extractor.rb

```diff
@@ -93,6 +87,19 @@ def instrument(type, field)
       field.redefine { resolve(new_resolve_proc) }
     end
 
+    def extract!(query)
+      @query = query
+
+      extract_query
+
+      query.operations.each_value do |operation|
```

Comment:
Thinking through something Scott mentioned... if `irep_selection` is already limited to the current operation, is it weird that we're extracting variables for *all* operations here?

Reply:
👀 I think you're right. I'll create an issue and some tests for this separately.

Reply:
For the general metrics use case I think we wanted to log the use of fields in ops that weren't used, since they could still cause validation errors? So maybe it's the use of `irep_selection` that is incorrect?

Reply:
We should probably ship this PR as is and deal with that separately anyway. It may even technically be a breaking change while this PR isn't.